addons/html_editor/static/src/core/history_plugin.js

```diff
@@ -264,56 +263,50 @@ export class HistoryPlugin extends Plugin {
         return step;
     }
 
-    enableObserver(initObserver = false) {
-        if (initObserver) {
-            this.isObserverInit = true;
-        } else {
-            // Don't count initial enableObserver, it is not accompanied by a disableObserver
-            this.ignoreDOMChangesDepth--;
-        }
-
-        if (this.isObserverInit && this.ignoreDOMChangesDepth === 0) {
-            this.observer.observe(this.editable, {
-                childList: true,
-                subtree: true,
-                attributes: true,
-                attributeOldValue: true,
-                characterData: true,
-                characterDataOldValue: true,
-            });
+    enableObserver() {
+        if (this.enableObserverCallbacks.size > 0) {
+            return;
         }
+        this.observer.observe(this.editable, {
+            childList: true,
+            subtree: true,
+            attributes: true,
+            attributeOldValue: true,
+            characterData: true,
+            characterDataOldValue: true,
+        });
     }
-
+    /**
+     * Disable the mutation observer.
+     *
+     * /!\ This method should be used with extreme caution. Not observing some
+     * mutations could lead to mutations that are impossible to undo/redo.
+     */
     disableObserver() {
-        if (this.isObserverInit && this.ignoreDOMChangesDepth === 0) {
-            this.handleObserverRecords();
-            this.observer.disconnect();
-        }
-        this.ignoreDOMChangesDepth++;
+        const enableObserver = () => {
+            this.enableObserverCallbacks.delete(enableObserver);
+            this.enableObserver();
+        };
+        this.enableObserverCallbacks.add(enableObserver);
+        this.handleObserverRecords();
+        this.observer.disconnect();
+        return enableObserver;
     }
 
     /**
-     * Execute {@link callback} while DOM observer is disabled.
+     * Execute {@link callback} while the MutationObserver is disabled.
+     *
+     * /!\ This method should be used with extreme caution. Not observing some
+     * mutations could lead to mutations that are impossible to undo/redo.
      *
      * @param {Function} callback
-     * @return {ReturnType<callback>}
      */
     ignoreDOMChanges(callback) {
-        this.disableObserver();
-        let result;
-        let isResultPromise;
+        const enableObserver = this.disableObserver();
         try {
-            result = callback();
-            isResultPromise = result?.finally;
-            return isResultPromise
-                ? result.finally(() => {
-                      this.enableObserver();
-                  })
-                : result;
+            Promise.resolve().then(callback()).finally(enableObserver);
         } finally {
-            if (!isResultPromise) {
-                this.enableObserver();
-            }
+            enableObserver();
```

Comment:
IMO, this calls enable twice

Reply:
Which shouldn't be a problem, except that the `finally { ... }` is invoked before the `.finally(...)`